torch/csrc/jit/codegen/cuda/ir_graphviz.h

```diff
@@ -0,0 +1,121 @@
+
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/codegen/cuda/dispatch.h>
+
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+// Corresponding unit test
+void testGPU_IrGraphGenerator();
+
+namespace fuser {
+
+// Generates a DOT (https://www.graphviz.org) graph
+// representation of a fuser IR
+//
+// Usage:
+// 1) Add calls to IrGraphGenerator::print(), for example:
+//  `IrGraphGenerator::print(&fusion, "ir.dot")`
+//
+// 2) Call IrGraphGenerator::print() from a debugger. Using gdb for example:
+//  `call IrGraphGenerator::print(&fusion, "ir.dot",
+//      IrGraphGenerator::DetailLevel::Explicit)`
+//
+// Notes:
+//  - When called from the debugger, the detail_level must be
+//    explicitly passed in (most debuggers don't support default arguments)
+//
+//  - The output dot file path can't include shell specific notations,
+//    for example you can't use "~/temp/ir.dot" ("/home/user/temp/ir.dot"
+//    must be used instead)
+//
+class TORCH_CUDA_API IrGraphGenerator : private OptInConstDispatch {
+  friend void torch::jit::testGPU_IrGraphGenerator();
+
+ public:
+  enum class DetailLevel {
+    ComputeOnly, // Only dataflow (compute) nodes
+    Basic, // Compute + schedule, with minimal details (default)
+    Explicit, // Additional details (ex. symbolic names for scalar constants)
+    Verbose, // Includes all values and dead definitions
+  };
+
+ public:
```

Comment:
nitpick `public:` here again.

Reply:
public is technically redundant here indeed, although I prefer to put member types in a separate section from member functions (and member data) sections.

is this frowned upon?

Reply:
If this is intentional, then totally ignore me.
I don't have any opinion on this and was pointing it out just in case it's an oversight.